### ISSUE TITLE
The SMPlayer window flicks and turns solid green color all the time

### ISSUE THREAD

**Issue by ElektroStudios** [smplayer-dev/smplayer#398]
**Describe the bug**

The problem is that the SMPlayer window turns completely green for an instant. The window flicks and renders entirely to green when changing the SMPlayer window size. 

Note that this only happens with MPlayer engine. Not with MPV. But since seeking a video it has always been a lot better / faster / responsive using MPlayer engine than MPV, then I'm not going to change to MPV engine just to get rid of this issue.

It happens with any kind of video format I tried, so I think its a MPlayer render issue rather than a decoding issue.

**To Reproduce**
Make double click in the SMPlayer window to change the window size, I mean to change between fullscreen mode and normal mode. You will see the window will turn green for a second every time you do this.

Or another common scenario: just make SMPlayer window fullscreen, and move the mouse cursor at the bottom of the screen to pop up the bar with seeking controls and etc. You will see the window will turn green for a second every time you do this, everytime that the bottom control bar appear and disappear.

**Screenshots**
![](https://i.imgur.com/DwC2rtN.png)

**Your Enviroment**
 - SMPlayer version: 22.2.0 x64
 - OS: Windows 10 x64
 - Multimedia engine you're using with SMPlayer: MPlayer

**Comment by ElektroStudios** [comment]
If helpful, let me say [smplayer-portable-21.10.0.0-x64.7z](https://www.videohelp.com/download/smplayer-portable-21.1.0.0-x64.7z) and prior versions (21.8, 21.1, 20.6, etc) does not have this bug.

This green render color issue is only present since release version 22.2.0.

I'm talking only about release builds, I didn't tried to find and test experimental or nightly builds.

**Comment by smplayer-dev** [comment]
What video output are you using? (preferences -> general -> video)
Do you still have the problem if you select another one?

**Comment by ElektroStudios** [comment]
> What video output are you using? (preferences -> general -> video) Do you still have the problem if you select another one?

Tried this:
![](https://i.imgur.com/3tlH8sj.png)

And this:
![](https://i.imgur.com/1llVnkb.png)

None of both configurations solves the problem.

I discovered that deleting the smplayer.ini, that is, restoring all values to default solves the issue. So maybe the problem is not under the video config values.

I will look a bit for what value in the ini file is causing this.

UPDATE:
Problem is in the [%General] ini section. Didn't yet discovered which value.

**Comment by ElektroStudios** [comment]
Ok I found that having a default smplayer.ini, that is open smplayer for first time and close it to let generate the default smplayer.ini file, if I change this value under [%General] ini section:

`mplayer_bin=mpv/mpv.exe`

To this:

`mplayer_bin=mplayer/mplayer.exe`

The problem of the green screen starts to be present.

Changing to mpv gets rid of the problem, but... I don't like to use mpv engine in smplayer (the video seeking is better performed with mplayer engine).

I tried replacing the entire mplayer folder, from smplayer v21.1.0 to v22.2.0. The green screen problem persists.

The mplayer rendering mechanics of v21.1.0 works fine for me with the same smplayer.ini config. And for any version prior to v22.2.0.

**Comment by ye110wd** [comment]
may not be related but same behaviour in linux deb with new version but with chunks of frame drops instead of green screen. Only vaapi does not do this and works.
xv, vdpau, x11 all have the problem, gpu intermittent.

xv has 100% load on single cpu core/thread. All others have normal load (1-3%)

the Default selected a frame drop Output Driver that had problem.

I downgraded to smplayer_21.10.0-1.ubuntu-focal_amd64.deb and now it works properly in full screen

edit> all above using mpv

**Comment by smplayer-dev** [comment]
I can't reproduce this problem in my computer. Could you test if the problem gets fixed if you change the video output from "default" to "direct3d" in preferences, general, video?

**Comment by ElektroStudios** [comment]
> Could you test if the problem gets fixed if you change the video output from "default" to "direct3d" in preferences, general, video?

Yes, the problem gets fixed changing to direct3d with mplayer engine.

I wonder why you can't reproduce the problem when in my case is so simple as changing mpv to mplayer engine. That mean the cause of the problem could be related to my graphics card or something else that would be near impossible to reproduce on your side?. In case of this info can help, I have a nVidia GTX 1060 with drivers updated, and I don't have installed any kind of codecs pack.

**Comment by smplayer-dev** [comment]
The mplayer's default video output is directx. I guess for some reason it's not working correctly in your computer.

**Comment by ElektroStudios** [comment]
> The mplayer's default video output is directx. I guess for some reason it's not working correctly in your computer.

I'm not understanding something...

I'm using SMPlayer v21.1.0 x64 because works fine without this issue neither the window position issue. And I'm using the default video output with MPlayer engine. It works fine. However, In that SMPlayer version (and in the other recent builds) if I change to DirectX (fast or slow) I get that green screen.

But you say MPlayer uses by default DirectX. Then why when I use SMPlayer with MPlayer and the default video output it works for me, but If I change to DirectX fast or slow I start to get this issue?. Maybe MPlayer of SMPlayer 21.1.0 default video output is not DirectX then?.

Anyways, you could think something that I could try to solve this issue with the latest SMPlayer build?. I have my graphics card drivers updated.

**Comment by smplayer-dev** [comment]
Previous versions of smplayer used direct3d by default. That was changed in version 22.2 because of [this request](https://github.com/smplayer-dev/smplayer/issues/349).
